crates/astria-sequencer-relayer/src/relayer/celestia_client/error.rs

```diff
@@ -0,0 +1,149 @@
+    BlobTooLarge { byte_count: usize },
+    /// The celestia app responded with the given error status to a `QueryAccountRequest`.
+    #[error("failed to get account info - is correct celestia signing key in use?")]
+    FailedToGetAccountInfo(#[source] GrpcResponseError),
+    /// The account info response was empty.
+    #[error("the account info response was empty")]
+    EmptyAccountInfo,
+    /// The account info response was of an unexpected type.
+    #[error("expected '{expected}' but received '{received}'")]
+    AccountInfoTypeMismatch { expected: String, received: String },
+    /// Failed to decode the received account info.
+    #[error("failed to decode account info")]
+    DecodeAccountInfo(#[source] ProtobufDecodeError),
+    /// The celestia app responded with the given error status to a `BroadcastTxRequest`.
+    #[error("failed to broadcast transaction")]
+    FailedToBroadcastTx(#[source] GrpcResponseError),
+    /// The broadcast transaction response was empty.
+    #[error("the broadcast transaction response was empty")]
+    EmptyBroadcastTxResponse,
+    /// The broadcasted transaction response contains an error code.
+    #[error(
+        "broadcast transaction response contains error code {code}, tx {tx_hash}, namespace \
+         '{namespace}', log: '{log}'"
+    )]
+    BroadcastTxResponseErrorCode {
+        tx_hash: String,
+        code: u32,
+        namespace: String,
+        log: String,
+    },
+    /// The celestia app responded with the given error status to a `GetTxRequest`.
+    #[error("failed to get transaction")]
+    FailedToGetTx(#[source] GrpcResponseError),
+    /// The get transaction response was empty.
+    #[error("the get transaction response was empty")]
+    EmptyGetTxResponse,
+    #[error(
+        "get transaction response contains error code {code}, tx {tx_hash}, namespace \
+         '{namespace}', log: {log}"
+    )]
+    GetTxResponseErrorCode {
+        tx_hash: String,
+        code: u32,
+        namespace: String,
+        log: String,
+    },
+}
+
+/// An error while encoding a Bech32 string.
+#[derive(Error, Clone, Debug)]
+#[error(transparent)]
+pub(in crate::relayer) struct Bech32EncodeError(#[from] bech32::EncodeError);
+
+/// A gRPC status representing an error response from an RPC call.
+#[derive(Clone, Debug)]
+pub(in crate::relayer) struct GrpcResponseError(Status);
+
+impl Display for GrpcResponseError {
+    fn fmt(&self, formatter: &mut Formatter<'_>) -> fmt::Result {
+        write!(
+            formatter,
+            "status: {}, message: {}, metadata: {:?}",
+            self.0.code(),
+            self.0.message(),
+            self.0.metadata(),
+        )
+    }
+}
+
+impl From<Status> for GrpcResponseError {
+    fn from(status: Status) -> Self {
+        Self(status)
+    }
+}
+
+impl std::error::Error for GrpcResponseError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        self.0.source()
+    }
+}
```

Comment:
This is almost equivalent to just decorating the `GrpcResponseError` definition with `#[error(transparent)]`.  However, [the `Display` impl for `Status`](https://github.com/hyperium/tonic/blob/v0.10.2/tonic/src/status.rs#L710-L721) delegates to the `Debug` impls of its member fields including the `details: Bytes` field and is overly verbose.  Hence I just hand-rolled these trait impls.